kube/src/config/file_config.rs

```diff
@@ -200,34 +202,27 @@ impl AuthInfo {
                         }
 
                         if let Some(field) = provider.config.get("token-key") {
-                             // incredbly naive JSONPath impl
-                            let splitted = field
-                                .trim_matches(|c| c == '"' || c == '{' || c == '}')
-                                .split("." )
-                                .filter(|a| a.len() > 0);
-                            let mut current : serde_json::Value = serde_json::from_slice(&output.stdout)?;
-                            for item in splitted {
-                                if let serde_json::Value::Object(mut map) = current {
-                                    if let Some(v) = map.remove(item) {
-                                        current = v;
+                            let pure_path = field
+                                .trim_matches(|c| c == '"' || c == '{' || c == '}');
```

Comment:
does this remove __all__ quotes and braces? doesn't that limit the jsonpath variants?

Reply:
[`trim_matches` only remove from left and right](https://doc.rust-lang.org/std/string/struct.String.html#method.trim_matches_) as long as they match. Any inner usage of them would be kept (other than what `replace` would do). [playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=75d323d603d87ef34ebd7be69507de00).